src/app/state/playback_state.rs

```diff
@@ -292,8 +295,10 @@ impl UpdatableState for PlaybackState {
             PlaybackAction::TogglePlay => {
                 if let Some(playing) = self.toggle_play() {
                     if playing {
+                        self.seek_position.resume();
```

Comment:
(applies to other lines below)

let's try to keep as much of the state manipulation inside the various methods above (toggle_play in that particular case) so that the already quite large match does not get larger!

Reply:
Excellent point! I've pushed a commit that hoists everything apart from seeking into the appropriate methods. 

(I'm gonna blame me being a fish out of water with Builder for not thinking of this :p)